src/runtime.js

```diff
@@ -196,6 +196,11 @@ var Runtime = {
 #if ASSERTIONS
       assert(args.length == sig.length-1);
 #endif
+      if (Module['usingWasm']) {
+        // look up function by name and call it directly
+        name = Module['indirectFunctions'][ptr]
+        return Module[name].apply(null, args);
```

Comment:
er, sorry that should be `Module['asm'][name].apply()` which means we are calling on the wasm export object instead of the Module object. But I guess those would be more or less equivalent lookups.